server/market/orderrouter.go

```diff
@@ -628,11 +628,7 @@ func (r *OrderRouter) sufficientAccountBalance(accountAddr string, ord order.Ord
 			fundingQty = trade.Quantity
 			fundingLots = trade.Quantity / tunnel.LotSize()
 		} else {
-			if lo, ok := ord.(*order.LimitOrder); ok {
-				redeems = int(calc.QuoteToBase(lo.Rate, trade.Quantity) / tunnel.LotSize())
-			} else {
-				redeems = int(calc.QuoteToBase(safeMidGap(tunnel), trade.Quantity) / tunnel.LotSize())
-			}
+			redeems = int(trade.Quantity / tunnel.LotSize())
```

Comment:
I think it should be
```Go
if _, isLimit := ord.(*order.LimitOrder); isLimit {
	redeems = int(trade.Quantity / tunnel.LotSize())
} else { // market buy
	redeems = int(calc.QuoteToBase(safeMidGap(tunnel), trade.Quantity) / tunnel.LotSize())
}
```

Reply:
Having trouble wrapping my head around this. I will take your word for it but...

Why not always the number of possible trades? redeems is the number of redeems, not amount of asset needed to redeem. So `QuoteToBase` as in the amount of redeem asset doesn't make sense to me.

Reply:
> redeems is the number of redeems, not amount of asset needed to redeem. So `QuoteToBase` as in the amount of redeem asset doesn't make sense to me.

It's still dividing by lot size there; the quantity just needs to be in base asset to determine lots.

I think this indicates some more docs in the method could be added.

```go
func (r *OrderRouter) sufficientAccountBalance(accountAddr string, ord order.Order, assetID, redeemAssetID uint32, tunnel MarketTunnel) bool {
	trade := ord.Trade()

	// This asset is funding an order when it is either:
	//  - base asset in a sell order e.g. selling ETH in a ETH-LTC market
	//  - quote asset in a buy order e.g. buying BTC in a BTC-ETH market
	// This asset will be redeemed when it is either:
	//  - base asset in a buy order e.g. buying ETH in a ETH-LTC market
	//  - quote asset in a sell order e.g. selling in a BTC-ETH market

	var fundingQty, fundingLots uint64 // when the asset is base in sell order, or quote in buy order
	var redeems int                    // when the asset is base in buy order, or quote in sell order
	if ord.Base() == assetID {
		if trade.Sell {
			fundingQty = trade.Quantity
			fundingLots = trade.Quantity / tunnel.LotSize()
		} else { // buying base asset
			baseQty := trade.Quantity
			if _, ok := ord.(*order.MarketOrder); ok {
				// Market buy Quantity is in units of quote asset, so estimate
				// how much of base asset that might be based on mid-gap rate.
				baseQty = calc.QuoteToBase(safeMidGap(tunnel), trade.Quantity)
			}
			redeems = int(baseQty / tunnel.LotSize())
		}
	} else { ...
```

Also note I've removed the input arg because nothing in the dex.Asset was really needed.  We need to stop passing big structs.

Reply:
Thank you for the comments. https://github.com/decred/dcrdex/compare/314a1d1dd3a01e28bf90b760f151a8ed242a8297..c02e5a418ae36fe81c8b579ae77136bc9a77a004